src/pages/ListSwaps/SwapRow.tsx

```diff
@@ -40,208 +29,52 @@ function AssetCell({ asset }: AssetCellProps) {
 
-function ledgerParamsReducer(currentState: object, field: LedgerParamsField) {
-  return { ...currentState, [field.name]: field.value };
-}
-
-function actionLedger(swap: Swap, action: string) {
-  switch (action) {
-    case "redeem":
-      return swap.role === "Alice"
-        ? swap.parameters.beta_ledger
-        : swap.parameters.alpha_ledger;
-    case "refund":
-    case "fund":
-      return swap.role === "Alice"
-        ? swap.parameters.alpha_ledger
-        : swap.parameters.beta_ledger;
-  }
-}
-
-function SwapRow(swap: Swap) {
-  const [dialogState, setDialogState] = useState(DialogState.Closed);
-  const [action, setAction] = useState({
-    name: "",
-    url: URI("")
-  });
-  const [ledgerActionParamSpec, setLedgerActionParamSpec] = useState<
-    LedgerActionSpec[]
-  >([]);
-  const initialLedgerActionParams = {} as { [key: string]: string };
-  const [ledgerActionParams, dispatchLedgerActionParams] = useReducer(
-    ledgerParamsReducer,
-    initialLedgerActionParams
+  const actions = actionDialogs(
+    swap.links,
+    swap.properties.role,
+    swap.properties.parameters.alpha_ledger,
+    swap.properties.parameters.beta_ledger
   );
 
-  const alphaIsTransitive = swap.parameters.alpha_ledger.name === "bitcoin";
-  const betaIsTransitive = swap.parameters.beta_ledger.name === "bitcoin";
-
-  const actionButtons = Object.entries(swap._links)
-    .filter(([key, _]) => key !== "self")
-    .map(([actionName, actionUrl]) => {
-      return (
-        <Button
-          variant="outlined"
-          color="primary"
-          key={actionName}
-          onClick={() =>
-            handleClickAction(actionName, apiEndpoint().path(actionUrl.href))
-          }
-        >
-          {actionName}
-        </Button>
-      );
-    });
-
-  const acceptFields = [];
-  if (!alphaIsTransitive) {
-    acceptFields.push({
-      key: "alpha_ledger_redeem_identity",
-      label: "Alpha Redeem Identity"
-    });
-  }
-  if (!betaIsTransitive) {
-    acceptFields.push({
-      key: "beta_ledger_refund_identity",
-      label: "Beta Refund Identity"
-    });
-  }
-
-  const handleClickAction = (name: string, url: uri.URI) => {
-    setAction({ name, url });
-
-    switch (name) {
-      case "accept":
-      case "decline":
-        setDialogState(DialogState.CommunicationDialogOpen);
-        break;
-      case "fund":
-      case "deploy":
-      case "redeem":
-      case "refund":
-        const params = actionQueryParams(swap, name);
-
-        if (params.length > 0) {
-          setLedgerActionParamSpec(params);
-          setDialogState(DialogState.LedgerParamsDialogOpen);
-        } else {
-          setDialogState(DialogState.LedgerDialogOpen);
-        }
-        break;
-    }
-  };
-
   return (
-    <React.Fragment key={swap._links.self.href}>
-      <TableRow data-cy="swap-row">
-        <TableCell>{swap.parameters.alpha_ledger.name}</TableCell>
+    <React.Fragment key={swap.links[0].href}>
```

Comment:
:mega: Please use the `self` link and not just the first one.

Reply:
Didn't notice that! I guess it's just a key, but I'll use `swapLink` instead.